exo-deploy/src/deployers/aws-deployer.ls

```diff
@@ -39,20 +41,62 @@ class AwsDeployer
 
   deploy: ->
     @terraform
-      ..get (err) ->
+      ..get (err) ~>
         | err => return process.stdout.write err.message
-        process.stdout.write "terraform starting deploy to AWS"
-        ..apply (err) ->
-          | err => return process.stdout.write err.message
+        @_get-hosted-zone-id destroy: no, (err, hosted-zone-id) ~>
+          | err => process.stdout.write "Cannot get hosted zone id #{err.message}" ; return
+          ..apply {hosted-zone-id}, (err) ->
+            | err => return process.stdout.write err.message
 
 
   nuke: ->
-    @terraform
-      ..get (err) ->
-        | err => return process.stdout.write err.message
-        process.stdout.write "terraform starting nuke from AWS"
-        ..destroy (err) ~>
-          | err => return process.stdout.write err.message
+    @terraform-file-builder.generate-provider-credentials!
+    process.stdout.write "removing the entire AWS deployment"
+    @_get-hosted-zone-id destroy: yes, (err, hosted-zone-id) ~>
+      | err => process.stdout.write "Cannot get hosted zone id #{err.message}" ; return
+      @terraform.destroy {hosted-zone-id}, (err) ~>
+          | err => process.stdout.write "Terraform cannot destroy infrastructure #{err.message}" ; return
+          @_remove-hosted-zone @domain-name, (err) ->
+            | err => process.stdout.write "Cannot remove hosted zone #{err.message}"
+
+
+  _get-hosted-zone-id: ({destroy}, done) ->
+    @_hosted-zone-exists @domain-name, (err, hosted-zone) ~>
+      | err         => process.stdout.write err.message ; return done err
+      | hosted-zone => done null, hosted-zone.Id
+      | _           =>
```

Comment:
```ls
@_hosted-zone-exists @domain-name, (err, hosted-zone) ~>
  | err          =>  process.stdout.write err.message ; return done err
  | hosted-zone  =>  done null, hosted-zone.Id
  | destroy      =>  done!
  | _            =>  @_create-hosted-zone @domain-name, done
```

The method `_create-hosted-zone` should print error messages related to creating the hosted zone, like when it cannot create it.